AdobeStockImage/Test/Integration/Model/SaveImageTest.php

```diff
@@ -0,0 +1,213 @@
+    {
+        $this->deleteImage($destinationPath);
+        $document = $this->getDocument($documentData);
+        $mediaDir = $this->fileSystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->saveImage->execute(
+            $document,
+            $this->getImageFilePath($sourceFile),
+            $mediaDir->getAbsolutePath($destinationPath)
+        );
+        self::assertTrue(
+            $this->fileSystem->getDirectoryRead(DirectoryList::MEDIA)->isExist($destinationPath),
+            'File was not saved by destination'
+        );
+        $searchCriteria = $this->criteriaBuilder
+            ->addFilter('creator_id', $document->getCustomAttribute('creator_id')->getValue(), 'eq')
+            ->create();
+        self::assertNotEmpty(
+            $this->assetRepository->getList($searchCriteria),
+            'Image asset was not saved'
+        );
+        $this->deleteImage($destinationPath);
+    }
+
+    /**
+     * Document for save.
+     *
+     * @param array $documentData
+     * @return Document
+     */
+    private function getDocument(array $documentData): Document
+    {
+        $document = new Document($documentData);
+        $this->addAttributes($document, $documentData['extension_attributes']);
+        return $document;
+    }
+
+    /**
+     * Add attributes to document
+     *
+     * @param Document $document
+     * @param array $attributes [code => value]
+     * @return Document
+     */
+    private function addAttributes(Document $document, array $attributes): Document
+    {
+        $customAttributes = $document->getCustomAttributes();
+        $attributeValueFactory = Bootstrap::getObjectManager()->create(
+            AttributeValueFactory::class
+        );
+        foreach ($attributes as $code => $value) {
+            $attribute = $attributeValueFactory->create();
+            $attribute->setAttributeCode($code);
+            $attribute->setValue($value);
+            $customAttributes[$code] = $attribute;
+        }
+
+        $document->setCustomAttributes($customAttributes);
+
+        return $document;
+    }
+
+    /**
+     * Return image file path
+     *
+     * @param string $filename
+     * @return string
+     */
+    private function getImageFilePath(string $filename): string
+    {
+        return dirname(__DIR__, 1)
+            . DIRECTORY_SEPARATOR
+            . implode(
+                DIRECTORY_SEPARATOR,
+                [
+                    '_files',
+                    $filename
+                ]
+            );
+    }
+
+    /**
+     * Delete test image if exists
+     *
+     * @param string $destinationPath
+     * @return void
+     */
+    private function deleteImage(string $destinationPath): void
+    {
+        $mediaDir = $this->fileSystem->getDirectoryWrite(DirectoryList::MEDIA);
+        if ($mediaDir->isExist($destinationPath)) {
+            $this->driver->deleteFile($mediaDir->getAbsolutePath($destinationPath));
+        }
```

Comment:
Delete method is present in the directory class, the isExist check is not necessary
```suggestion
                    $this->fileSystem->getDirectoryWrite(DirectoryList::MEDIA)->delete($destinationPath);
```